src/backend/distributed/executor/multi_utility.c

```diff
@@ -3770,3 +3776,54 @@ PlanGrantStmt(GrantStmt *grantStmt)
 
 	return ddlJobs;
 }
+
+
+/*
+ * ProcessDropTableStmt processes DROP TABLE commands for partitioned tables.
+ * If we are trying to DROP partitioned tables, we first need to go to MX nodes
+ * and DETACH partitions from their parents. Otherwise, we process DROP command
+ * multiple times in MX workers. For shards, we send DROP commands with IF EXISTS
+ * parameter which solves problem of processing same command multiple times.
+ * However, for distributed table itself, we directly remove related table from
+ * Postgres catalogs via performDeletion function, thus we need to be cautious
+ * about not processing same DROP command twice.
+ */
+static void
+ProcessDropTableStmt(DropStmt *dropTableStatement)
+{
+	ListCell *dropTableCell = NULL;
+
+	Assert(dropTableStatement->removeType == OBJECT_TABLE);
+
+	foreach(dropTableCell, dropTableStatement->objects)
+	{
+		List *tableNameList = (List *) lfirst(dropTableCell);
+		RangeVar *tableRangeVar = makeRangeVarFromNameList(tableNameList);
+		bool missingOK = true;
+		List *partitionList = NIL;
+		ListCell *partitionCell = NULL;
+
+		Oid relationId = RangeVarGetRelid(tableRangeVar, AccessExclusiveLock, missingOK);
+
+		if (relationId == InvalidOid ||
+			!IsDistributedTable(relationId) ||
+			!ShouldSyncTableMetadata(relationId) ||
+			!PartitionedTable(relationId))
+		{
+			continue;
+		}
+
+		partitionList = PartitionList(relationId);
```

Comment:
Shouldn't we return if `partitionList` has zero entries?

Reply:
DROP TABLE query can have multiple tables in it. partitionList may be empty for first table but it may have entries for next ones. Still, I added an if check to `continue` so that we will not send lots of unnecessary commands to disable DDL propagation. (We still send some unnecessary ones unfortunately. I can add an additional flag to prevent that but I guess that would be overkill and cause difficult to maintain code.)

Reply:
We don't have to do it, but, I don't think it'd bad to add a flag named `sentDisableDDL` and skip it if already `true`.